File: 00OOPS/01Basics.cpp
#include <iostream>
using namespace std;

class Student
{
public:
    string name;
    string grade;
    int age;
    int roll_no;
};

int main()
{

    Student s1;
    s1.name = "Rohan";
    s1.grade = "A+";
    s1.age = 20;
    s1.roll_no = 30;

    Student s2;
    s2.name = "Mohan";
    s2.grade = "A+";
    s2.age = 22;
    s2.roll_no = 31;
}

File: OOPS-TS/src/01Basics.ts
// class Student {
//     name!: string
//     grade!: string
//     age!: number
//     roll_no!: number

// }

// let s1 = new Student()
// s1.name = "Abhi"


// --------------------------------------

class Student {
    name: string
    grade: string
    age: number
    roll_no: number

    constructor(name: string, grade: string, age: number, roll_number: number) {
        this.name = name
        this.age = age
        this.grade = grade
        this.roll_no = roll_number
    }

}

const studentOne = new Student("Rohan", "A+", 20, 12)
console.log(studentOne)
console.log(studentOne.name)

// ------------------------------------------------------------